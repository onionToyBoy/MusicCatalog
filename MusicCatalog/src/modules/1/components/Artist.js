import React from 'react';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Text,
} from 'react-native';

import {colors} from '../../../constants/colors';
import {symbols} from '../../../constants/symbols';

export const Artist = ({name, genre, openAlbums}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={() => openAlbums(name)}>
      <View style={styles.leftSide}>
        <Text style={styles.name}>{name}</Text>
        <Text style={styles.gerne}>{genre}</Text>
      </View>
      <View style={styles.rightSIde}>
        <Text style={styles.bracket}>{symbols.RIGHT_ANGULAR_BRACKET}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 10,
    backgroundColor: colors.MIDDLE_GRAY,
    borderBottomColor: colors.BRIGHT_GRAY,
    borderBottomWidth: 2,
  },
  name: {
    color: colors.WHITE,
    fontSize: 20,
  },
  gerne: {
    color: colors.GOLD,
    fontSize: 15,
  },
  bracket: {
    color: colors.BRIGHT_GRAY,
    fontSize: 35,
  },
  rightSIde: {
    flex: 1,
    justifyContent: 'center',
    alignItems:'flex-end',
  },
  leftSide: {
    flex: 1,
  },
});
