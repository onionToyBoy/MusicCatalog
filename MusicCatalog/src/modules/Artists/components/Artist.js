import React from 'react';
import { View, StyleSheet, TouchableOpacity, Text } from 'react-native';

import { colors } from '../../../constants/colors';
import { symbols } from '../../../constants/symbols';

export const Artist = ({ name, genre, openAlbums, id,componentId }) => {
  return (
    <TouchableOpacity style={styles.container} onPress={() => openAlbums(componentId,name, id)}>
      <View style={styles.leftSide}>
        <Text style={styles.name}>{name}</Text>
        <Text style={styles.gerne}>{genre}</Text>
      </View>
      <View style={styles.rightSIde}>
        <Text style={styles.bracket}>{symbols.RIGHT_ANGULAR_BRACKET}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 10,
    backgroundColor: colors.MIDDLE_GRAY,
    borderBottomColor: colors.BRIGHT_GRAY,
    borderBottomWidth: 2,
  },
  name: {
    color: colors.WHITE,
    fontSize: 20,
  },
  gerne: {
    color: colors.GOLD,
    fontSize: 15,
  },
  bracket: {
    color: colors.BRIGHT_GRAY,
    fontSize: 30,
    marginRight: 3,
  },
  rightSIde: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
  leftSide: {
    flex: 1,
  },
});
