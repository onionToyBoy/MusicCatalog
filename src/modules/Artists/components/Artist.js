import React from 'react';
import { View, StyleSheet, TouchableOpacity, Text } from 'react-native';

import { colors } from '../../../constants/colors';
import { symbols } from '../../../constants/symbols';

export const Artist = ({ artistName, primaryGenreName, onOpenAlbum, artistId }) => {
  const onPressArtist = () => onOpenAlbum(artistName, artistId);

  return (
    <TouchableOpacity style={styles.container} onPress={onPressArtist}>
      <View style={styles.leftContainer}>
        <Text testID={'name'} style={styles.name}>
          {artistName}
        </Text>
        <Text testID={'genre'} style={styles.genre}>
          {primaryGenreName}
        </Text>
      </View>
      <View style={styles.rightContainer}>
        <Text style={styles.bracket}>{symbols.RIGHT_ANGULAR_BRACKET}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 10,
    backgroundColor: colors.MIDDLE_GRAY,
    borderBottomColor: colors.BRIGHT_GRAY,
    borderBottomWidth: 2,
  },
  name: {
    color: colors.WHITE,
    fontSize: 20,
  },
  genre: {
    color: colors.GOLD,
    fontSize: 15,
  },
  bracket: {
    color: colors.BRIGHT_GRAY,
    fontSize: 30,
    marginRight: 3,
  },
  rightContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
  leftContainer: {
    flex: 1,
  },
});
