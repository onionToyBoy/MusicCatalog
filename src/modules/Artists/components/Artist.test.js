import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Artist } from './Artist';

configure({ adapter: new Adapter() });

describe('Artist test', () => {
  const props = {
    artistName: 'Deftones',
    primaryGenreName: 'Rock',
    onOpenAlbum() {},
    artistId: 123,
  };

  test('Props in Artist component should be correct', () => {
    const component = shallow(<Artist {...props} />);

    //expect(component.props().onPress).toEqual(onPressArtist);
    expect(component.find('Text').at(0).props().children).toEqual(props.artistName);
    expect(component.find('Text').at(1).props().children).toEqual(props.primaryGenreName);
    expect(component).toMatchSnapshot();
  });
});
