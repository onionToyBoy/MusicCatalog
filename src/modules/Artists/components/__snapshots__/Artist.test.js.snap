// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist test  1`] = `ShallowWrapper {}`;

exports[`Artist test Props in Artist component should be correct 1`] = `ShallowWrapper {}`;
